scripts/ci/lib.sh

```diff
@@ -156,6 +156,46 @@ get_central_debug_dump() {
     ls -l "${output_dir}"
 }
 
+process_central_metrics() {
+    info "Processing metrics from central debug dump"
+
+    if [[ "$#" -ne 1 ]]; then
+        die "missing arg. usage: process_central_metrics <debug_dump_dir>"
+    fi
+
+    local metrics_output
+    local csv_output
+    metrics_output="$(mktemp --suffix=.prom)"
+    csv_output="$(mktemp --suffix=.csv)"
+    unzip -p "${output_dir}/*.zip" metrics-1 > "${metrics_output}"
+
+    get_prometheus_metrics_parser
+
+    # We need a link to repository. In case it's not part of job spec (e.g., periodic`s)
+    # we will fallback to short commit
+    base_link="$(echo "$JOB_SPEC" | jq ".refs.base_link | select( . != null )" -r)"
+    calculated_base_link="https://github.com/stackrox/stackrox/commit/$(make --quiet --no-print-directory shortcommit)"
+
+    echo "metric_name,labels,value,build_tag,build_id,orchestrator_flavor,job_name,base_link" > "${csv_output}"
+    prometheus-metric-parser single --format cvs --file "${metrics_output}" \
+      | awk -F"," \
+      -v build_tag="${STACKROX_BUILD_TAG}" \
+      -v build_id="${BUILD_ID}" \
+      -v orchestrator="${ORCHESTRATOR_FLAVOR:-PROW}" \
+      -v job_name="${JOB_NAME:-missing}" \
+      -v base_link="${base_link:-$calculated_base_link}" \
+      'BEGIN { OFS = "," } {$4=build_tag;$5=build_id;$6=orchestrator;$7=job_name;$7=base_link; print}' >> "${csv_output}"
```

Comment:
Thinking aloud:
if simple join with commas was insufficient [here](https://github.com/stackrox/prometheus-metric-parser/pull/15/files), then simple split and join on/with commas is likely to be fragile here too, no?

Reply:
Good point I can replace it with sed and it should be much simpler.
Unfortunately I've just discovered this is not working for multiline csv rows :(